metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -1408,8 +1410,108 @@ def visit_join_to_time_spine_node(self, node: JoinToTimeSpineNode) -> SqlDataSet
+            "This indicates internal misconfiguration."
+        )
+
+        # Build join expression.
+        time_spine_alias = self._next_unique_table_alias()
+        custom_granularity_name = node.time_dimension_spec.time_granularity.name
+        time_spine_source = self._get_time_spine_for_custom_granularity(custom_granularity_name)
+        left_expr_for_join: SqlExpressionNode = SqlColumnReferenceExpression.from_table_and_column_names(
+            table_alias=parent_alias, column_name=parent_time_dimension_instance.associated_column.column_name
+        )
+        if parent_time_dimension_instance.spec.time_granularity.base_granularity != time_spine_source.base_granularity:
+            # If needed, apply DATE_TRUNC to parent column match the time spine spine that's column being joined to.
+            left_expr_for_join = SqlDateTruncExpression.create(
+                time_granularity=time_spine_source.base_granularity, arg=left_expr_for_join
+            )
+        join_description = SqlJoinDescription(
+            right_source=SqlTableNode.create(sql_table=time_spine_source.spine_table),
+            right_source_alias=time_spine_alias,
+            on_condition=SqlComparisonExpression.create(
+                left_expr=left_expr_for_join,
+                comparison=SqlComparison.EQUALS,
+                right_expr=SqlColumnReferenceExpression.from_table_and_column_names(
+                    table_alias=time_spine_alias, column_name=time_spine_source.base_column
+                ),
+            ),
+            join_type=SqlJoinType.LEFT_OUTER,
+        )
+
+        # Remove base grain from parent dataset, unless that grain was also requested (in addition to the custom grain).
+        parent_instance_set = parent_data_set.instance_set
+        parent_select_columns = parent_data_set.checked_sql_select_node.select_columns
+        if not node.include_base_grain:
+            parent_instance_set = parent_instance_set.transform(
+                FilterElements(
+                    exclude_specs=InstanceSpecSet(time_dimension_specs=(parent_time_dimension_instance.spec,))
+                )
+            )
+            parent_select_columns = tuple(
+                [
+                    column
+                    for column in parent_select_columns
+                    if column.column_alias != parent_time_dimension_instance.associated_column.column_name
+                ]
+            )
+
+        # Build output time spine instances and columns.
+        time_spine_instance = TimeDimensionInstance(
+            defined_from=parent_time_dimension_instance.defined_from,
+            associated_columns=(self._column_association_resolver.resolve_spec(node.time_dimension_spec),),
+            spec=node.time_dimension_spec,
+        )
+        time_spine_instance_set = InstanceSet(time_dimension_instances=(time_spine_instance,))
+        time_spine_select_columns = (
```

Comment:
This only does one, so if I request `metric_time__martian_year` and `metric_time__martian_quarter` we'll do multiple joins. I think that's ok for now, we can optimize it later if we feel like it.